src/main/java/com/faforever/client/player/PlayerServiceImpl.java

```diff
@@ -40,11 +62,52 @@ void init() {
     lobbyServerAccessor.setOnPlayerInfoMessageListener(this);
     lobbyServerAccessor.setOnFoeListListener(this);
     lobbyServerAccessor.setOnFriendListListener(this);
+    gameService.addOnGameInfoBeanListener(change -> {
+      while (change.next()) {
+        for (GameInfoBean gameInfoBean : change.getRemoved()) {
+          gameInfoBean.setStatus(GameState.CLOSED);
+          gameInfoBean.getTeams().forEach((team, players) -> updatePlayerInfoBean(players, gameInfoBean));
```

Comment:
That's a common performance optimization. This way the method gets called 3 times (here and 2 times below). While it doesn't *really* matter for simple getters, it's a good practice in general since you never know how much work the method actually doing, or will do in future.